### DISCUSSION TITLE
WhatsApp Component Integration

### DISCUSSION THREAD

**Discussion by VelSankar-DT** [apache/camel-karavan#1059] [Q&A]
I have tried to integrate the WhatsApp component. But I am getting the following error

`2024-01-10 12:04:15.466 ERROR 14 --- [orker-thread-10] or.errorhandler.DefaultErrorHandler : Failed delivery for (MessageId: E1B1EB25E6C5516-0000000000000005 on ExchangeId: E1B1EB25E6C5516-0000000000000005). Exhausted after delivery attempt: 1 caught: java.lang.IllegalArgumentException: message must be specified
Message History (source location is disabled)
---------------------------------------------------------------------------------------------------------------------------------------
Source                                   ID                             Processor                                          Elapsed (ms)
                                        post-846e/post-846e            from[platform-http:///send?httpMethodRestrict=POST            6
internal-route.camel.yaml:6              post-846e/to5                  direct:whatsapp-direct                                        0
internal-route.camel.yaml:16             route-48af/log-27c3            log                                                           0
internal-route.camel.yaml:19             route-48af/setBody-3e16        setBody[simple{{
   "messaging_product": "whatsap            1
internal-route.camel.yaml:59             route-48af/log-e778            log                                                           1
internal-route.camel.yaml:62             route-48af/to-f9a8             whatsapp:168321619705814?authorizationToken=xxxxxx            0
Stacktrace
---------------------------------------------------------------------------------------------------------------------------------------
java.lang.IllegalArgumentException: message must be specified`

**YAML file:**

`- rest:
    id: rest-2c90
    post:
      - id: post-846e
        path: /send
        to: direct:whatsapp-direct
- route:
    id: route-48af
    nodePrefixId: route-d70
    from:
      id: from-9623
      uri: direct
      parameters:
        name: whatsapp-direct
      steps:
        - log:
            id: log-27c3
            message: ${body}
        - setBody:
            id: setBody-3e16
            expression:
              simple:
                id: simple-04e7
                expression: |-
                  {
                      "messaging_product": "whatsapp",
                      "to": "${jq(.customerMobileNumber)}",
                      "type": "template",
                      "template": {
                      "name": "e_receipt_template",
                      "language": {
                        "code": "en_US"
                      },
                      "components": [
                          {
                          "type": "header",
                          "parameters": [
                            {
                              "type": "document",
                              "document": {
                                "link": "${jq(.fileLink)}",
                                "filename": "${jq(.fileName)}"
                              }
                            }
                          ]
                        },
                        {
                          "type": "body",
                          "parameters": [
                            {
                              "type": "text",
                              "text": "name"
                            }
                          ]
                        }
                      ]
                    }
                  }
        - log:
            id: log-e778
            message: ${body}
        - to:
            id: to-f9a8
            uri: whatsapp
            parameters:
              phoneNumberId: '**************'
              authorizationToken: >-
                *****************
`

**Note:**
LogId: log-e778 is printed my entire message. But still getting "message must be specified" error.

**Comment by davsclaus** [comment]
messy bug report, hard to help when user report like that
